### PR TITLE
Added support for enum definitions. Improves usage of reusable enums.

### PR DESCRIPTION
**Pull request by wvpelt**
I have added the generation of enum definition types.
Empty files were generated for enum types.

### CODE REVIEW

**Review by peterfortuin — DISMISSED** [review]

**Comment by fazpu** [comment]
Thank you for the PR!

Please add a suitable example to the demo-app/client/swagger-files/unit-test-swagger.json and generate the test files. Thanks!

**Comment by yluijten** [comment]
> Thank you for the PR!
> 
> Please add a suitable example to the demo-app/client/swagger-files/unit-test-swagger.json and generate the test files. Thanks!

@fazpu Added gender in the UserDetails definition as an example enum in the demo-app.

**Review by peterfortuin — DISMISSED** [review]
Looks good. Awesome!

**Comment by wvpelt** [comment]
@fazpu Is it possible to merge this?

File: dist/common/templates/utils.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+export function safeSetValue(control: AbstractControl, value: any) {
+  if (control instanceof FormControl) {
+    control.setValue(value, {emitEvent: false});
+    return;
+  }
+
+  if (control instanceof FormArrayExtended) {
+    if (nullOrUndef(value)) value = [];
+    if (!Array.isArray(value)) throw new TypeError(`Cannot set value '${value}' on FormArrayExtended`);
```
Comment (mgamsjager) [inline dist/common/templates/utils.ts]: Bit inconsistent with the single line `if` statements (eg. line 29).

**Review by mgamsjager — APPROVED** [review]

**Review by mgamsjager — DISMISSED** [review]

**Comment by KnutRyagerInmeta** [comment]
I have some issues with this fork compared to v4.1.0. Other than the enums, it renames some methods, such that they conflict. I have the GET methods `approval` and `approval/count`, before the generated names were `approval` and `count`. Now they are both "getItems", which they share in the OpenApi definition, but no good to use due to conflict.

Also, for the enums, you can simply cast them `as const` instead of the enum type. This give them stricter type safety with less effort.

**Comment by jnwltr** [comment]
@wvpelt sorry for making you wait so long, and thanks a lot for th PR.
Anyway, if you were able to rebase, I would do the review and merge in a few days. Thanks.

**Comment by petrhavel2** [comment]
Hi, we are closing this PR because rebase was not done.

**Pull request closed without merging**